core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -335,7 +336,6 @@ class KafkaController(val config : KafkaConfig, zkUtils: ZkUtils, val brokerStat
       maybeTriggerPartitionReassignment()
       maybeTriggerPreferredReplicaElection()
       /* send partition leadership info to all live brokers */
-      sendUpdateMetadataRequest(controllerContext.liveOrShuttingDownBrokerIds.toSeq)
```

Comment:
@becketqin I think we don't need this. When controller start replica statement and partition and partition state machine, it should send LeanderAndIsrRequest for every partition that will have a leader. `addLeaderAndIsrRequestForBrokers()` will in turn call `addUpdateMetadataRequestForBrokers` to send MetadataUpdateRequest to all live brokers for these partitions. For partition that doesn't have a leader due to NoReplicaOnlineException, MetadataUpdateRequest is not needed for this partition either.